engine/robot.py

```diff
@@ -330,8 +374,97 @@ def set_phase(self, new_phase):
         with open(CSV_PATH + '/phases.csv', 'a') as fd:
             fd.write(str(self.phase) + '\n')
 
-    def execute_avoid_obstacle(self):
-        # TODO: SET BACK TO ORIGINAL MISSION (TRAVERSE OR RETURN)
+    def track_obstacle(self):
+        # make this async so main algorithm keeps running
+        # assuming front sensor is mounted in the center x position (width/2)
+        condition = True
```

Comment:
This condition was added because of testing. If its the real robot, we want to continuously check for obstacles. If its a simulation, we dont want a while loop because there's a finite mock sensor data

Reply:
not sure how i would rename it beside smthg like "want_to_get_sensor_data" or smthg

Reply:
so if its on our real robot it should be True and otherwise False?

Maybe we can use robot.is_sim?

Reply:
making try except and reducing condition to true